src/index.ts

```diff
@@ -88,18 +90,61 @@ const SchemaExtensionPlugin = makeExtendSchemaPlugin(build => {
             if (!representation || typeof representation !== "object") {
               throw new Error("Invalid representation");
             }
-            const { __typename, [nodeIdFieldName]: nodeId } = representation;
-            if (!__typename || typeof nodeId !== "string") {
-              throw new Error("Failed to interpret representation");
+
+            const { __typename } = representation;
+            if (!__typename) {
+              throw new Error(
+                "Failed to interpret representation, no typename"
+              );
+            }
+
+            if (nodeIdFieldName in representation) {
+              const { [nodeIdFieldName]: nodeId } = representation;
+              if (typeof nodeId !== "string") {
+                throw new Error(
+                  "Failed to interpret representation, invalid nodeId"
+                );
+              }
+
+              return resolveNode(
+                nodeId,
+                build,
+                fieldContext,
+                data,
+                context,
+                resolveInfo
+              );
+            } else {
+              // This only works with NodePlugin enabled
+              if (!getNodeIdForTypeAndIdentifiers) {
+                throw new Error("Failed to resolve node by identifiers");
+              }
+
+              const type = getTypeByName(__typename);
+              const {
+                pgIntrospection: {
+                  primaryKeyConstraint: { keyAttributes: attrs },
+                },
+              } = scopeByType.get(type);
+              const identifiers = attrs.map(
+                attr => representation[inflection.column(attr)]
+              );
```

Comment:
Looks good until here, but from here on we want to query in a way that will work even if we don't have the NodePlugin loaded. You should have access to `resolveInfo.graphile.selectGraphQLResultsFromTable` which you can read about here:

https://www.graphile.org/postgraphile/make-extend-schema-plugin/#the-selectgraphqlresultfromtable-helper